packages/lwc-language-server/src/metadata-utils/static-resources-util.ts

```diff
@@ -19,60 +20,91 @@ function getResourceName(resourceMetaFile: string) {
     return parse(resourceFile).name;
 }
 
-export async function updateStaticResourceIndex(updatedFiles: FileEvent[], { workspaceRoots }: WorkspaceContext, writeConfigs: boolean = true) {
+export async function updateStaticResourceIndex(updates: FileEvent[], { workspaceRoots }: WorkspaceContext, writeConfigs: boolean = true) {
     let didChange = false;
-    for (const f of updatedFiles) {
-        if (f.uri.endsWith('.resource-meta.xml')) {
-            if (f.type === FileChangeType.Created) {
-                didChange = true;
-                STATIC_RESOURCES.add(getResourceName(f.uri));
-            } else if (f.type === FileChangeType.Deleted) {
-                STATIC_RESOURCES.delete(getResourceName(f.uri));
-                didChange = true;
+
+    for (const update of updates) {
+        if (update.uri.endsWith('.resource-meta.xml')) {
+            const resourceName = getResourceName(update.uri);
+
+            switch (update.type) {
+                case FileChangeType.Created:
+                    didChange = true;
+                    STATIC_RESOURCES.add(resourceName);
+                case FileChangeType.Deleted:
+                    STATIC_RESOURCES.delete(resourceName);
+                    didChange = true;
             }
         }
-    }
-    if (didChange) {
-        return processStaticResources(workspaceRoots[0], writeConfigs);
+        if (didChange) {
+            return processStaticResources(workspaceRoots[0], writeConfigs);
+        }
     }
 }
 
 async function processStaticResources(workspace: string, writeConfigs: boolean): Promise<void> {
     if (STATIC_RESOURCES.size > 0 && writeConfigs) {
```

Comment:
@mrmicahcooper Is this logic accurate for the case where STATIC_RESOURCES size changes from 1 to 0?

Reply:
Hmm we didn't change that logic. This is how it was previously written. That being said, if you did delete all of your static resources, it does seem like this would not trigger for the last remaining resource.

I'll make a note to revisit if that's ok.

Reply:
@mrmicahcooper Yes, that sounds good, thanks.